app/src/androidTest/java/com/lucas/medicaltools/service/MedicalToolsInteractorTest.kt

```diff
@@ -0,0 +1,93 @@
+package com.lucas.medicaltools.service
+
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medical_equip.service.APIService
+import com.lucas.medical_equip.service.RetrofitBuilder
+import com.lucas.medicaltools.MainViewState
+import com.lucas.medicaltools.repository.MedicalToolsRepository
+import io.reactivex.Observable
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.mockito.ArgumentMatchers
+
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(JUnit4::class)
+class MedicalToolsInteractorTest {
```

Comment:
Item to test:
1. The success state flow.
2. The failure state flow.
3. Try to ensure all the states published in a flow is in the correct order. e.g. if you did the fetch medical tools intent, that it firsts has a loading state, then a rendered state.
4. Test the model is mapping correctly, all the fields etc and that it looks like you expect.